cmd/ui.go

```diff
@@ -330,12 +333,20 @@ func showProgress(
 			outMutex.Unlock()
 			return
 		case <-winch:
-			// More responsive progress bar resizing on platforms with SIGWINCH (*nix)
-			termWidth, _, _ = terminal.GetSize(fd)
+			if !errTermGetSize {
+				// More responsive progress bar resizing on platforms with SIGWINCH (*nix)
+				termWidth, _, err = terminal.GetSize(fd)
+				if err != nil {
+					termWidth = defaultTermWidth
+				}
+			}
 		case <-ticker.C:
 			// Default ticker-based progress bar resizing
-			if winch == nil {
-				termWidth, _, _ = terminal.GetSize(fd)
+			if !errTermGetSize && winch == nil {
```

Comment:
I still think that instead of here is a check for `winch`, we should
1. just call `ticker.Stop()` if `winch` won't be `nil` 
2. or for there to be 1 channel which is set to either `winch` or `ticker.C` depending on whether `winch` is supported.

Reply:
and if we are going to just not get the terminal size after the first error we can just as well set that channel to `nil` after the error and stop the ticker

Reply:
The thing is that we can't stop the ticker as that's what drives the progress bar refresh, so even if `winch != nil` (e.g. on Linux) both channels should be active. The `winch` one just makes the resizing more responsive.